models/Version/Adapter/DatabaseVersionStorageAdapter.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\Version\Adapter;
+
+use Doctrine\DBAL;
+
+final class DatabaseVersionStorageAdapter implements VersionStorageAdapterInterface
+{
+    CONST versionsTableName = "versionsData";
```

Comment:
should we automatically create the table it not existing? or do we really put that into the hands of the developer who sets it up?

Reply:
This we could do by catching `TableNotFoundException` and then running the DDL. 
But we need to be careful: we need to rollback the active db transaction, then run the CREATE TABLE and then throw an exception, so that the transaction get's restarted.